### PR TITLE
Create population pyramid for deceased people

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard10**
## Summary

In this PR we'll add the deceased age demographic chart. The existing `<AgeDemographic />`-component has been made generic and is now used for both the previous implementation as this new one.

image

The chart can now receive a prop called `visuallyMaxPercentage` which will clip large values. This way we can have a more readable chart when there's a big difference between the tallest bar and smallest bar.

Here's an example with `visuallyMaxPercentage` set to `30`:
image

### CODE REVIEW

File: packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts [hunk 1]
```diff
@@ -21,44 +22,65 @@ export interface AgeDemographicCoordinates {
   ageGroupPercentageScale: ValueOf<ScaleTypeToD3Scale<any, any, any>>;
   infectedPercentageScale: ValueOf<ScaleTypeToD3Scale<any, any, any>>;
   ageGroupRangeScale: ScaleBand<string>;
-  ageGroupPercentagePoint: (value: NationalTestedPerAgeGroupValue) => any;
-  infectedPercentagePoint: (value: NationalTestedPerAgeGroupValue) => any;
-  ageGroupRangePoint: (value: NationalTestedPerAgeGroupValue) => any;
-  getTooltipCoordinates: GetTooltipCoordinates<NationalTestedPerAgeGroupValue>;
+  ageGroupPercentagePoint: (value: T) => any;
+  infectedPercentagePoint: (value: T) => any;
+  ageGroupRangePoint: (value: T) => any;
```
Comment (VWSCoronaDashboard1) [inline packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts:27]: I'm assuming the `any` returns are inevitable here?

**Review by VWSCoronaDashboard4 — APPROVED** [review]
Nice!

**Review by VWSCoronaDashboard1 — APPROVED** [review]
This is a pretty complex refactor. It's quite hard to read through the changes, but all in all, this looks pretty damned good.

File: packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts (hunk 1 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts:27]: yeah I think so, those lines have only been moved so I didn't really dove into the typing.

File: packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx [hunk 2]
```diff
@@ -2,74 +2,93 @@ import css from '@styled-system/css';
...
-    event: MouseEvent<SVGElement>
-  ) => void;
+interface AgeDemographicChartProps<T extends AgeDemographicDefaultValue> {
+  coordinates: AgeDemographicCoordinates<T>;
+  text: AgeDemographicChartText;
+  onMouseMoveBar: (value: T, event: MouseEvent<SVGElement>) => void;
   onMouseLeaveBar: () => void;
   onKeyInput: (event: KeyboardEvent<SVGElement>) => void;
+  metricProperty: keyof T;
+  visuallyMaxPercentage?: number;
```
Comment (VWSCoronaDashboard8) [inline packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx:27]:
```suggestion
  visualMaxPercentage?: number;
```
I feel that is more correct English. Another suggestion is `displayMaxPercentage`

File: packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx [hunk 3]
```diff
@@ -124,9 +143,41 @@ export const AgeDemographicChart = memo<AgeDemographicChartProps>(
           stroke={colors.border}
         />
 
+        <PatternLines
+          id="is-clipped-pattern-age-range"
+          height={6}
+          width={6}
+          stroke={colors.data.neutral}
+          strokeWidth={2}
+          orientation={['diagonalRightToLeft']}
```
Comment (VWSCoronaDashboard8) [inline packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx:152]: Is there a scenario where you would pass multiple values here?

File: packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx [hunk 4]
```diff
@@ -209,3 +273,12 @@ const StyledHoverBar = styled(Bar)(
     },
   })
 );
+
+function getIsClipped(
+  value: number,
+  visuallyMaxPercentage: number | undefined
```
Comment (VWSCoronaDashboard8) [inline packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx:279]:
```suggestion
  visualMaxPercentage?: number
```

File: packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts [hunk 5]
```diff
@@ -21,44 +22,65 @@ export interface AgeDemographicCoordinates {
...
+function calculateAgeDemographicCoordinates<
+  T extends AgeDemographicDefaultValue
+>(
+  data: { values: T[] },
+  metricProperty: keyof T,
   isSmallScreen: boolean,
   parentWidth: number,
-  isExtraSmallScreen: boolean
-): AgeDemographicCoordinates {
+  isExtraSmallScreen: boolean,
```
Comment (VWSCoronaDashboard8) [inline packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts:81]: nit; 
```suggestion
  isTinyScreen: boolean,
```

Don't know, I just like the word :)

**Review by VWSCoronaDashboard8 — APPROVED** [review]

File: packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx (hunk 2 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx:27]: how about `clipValueAt`?

File: packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx (hunk 3 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx:152]: No idea, but this API is defined by visx :-)

File: packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts (hunk 5 above)
Comment (VWSCoronaDashboard1) [reply packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts:81]: And how do you feel about 'micro'? ;)

File: packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx (hunk 2 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components-styled/age-demographic/age-demographic-chart.tsx:27]: I'll go for `displayMaxPercentage`.

File: packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts (hunk 5 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts:81]: I'll stick to extraSmall becasue it's based on the breakpoints.xs :-)

Comment (VWSCoronaDashboard8) [reply packages/app/src/components-styled/age-demographic/age-demographic-coordinates.ts:81]: Hmm less exiting to me,  but I see you are going for same character count 😄 OCD. I'm fine with that too, although I don't think you would normally call the screens that we are talking about "micro" in real life.

**Pull request merged**